src/Utils/Namespaces/NamespaceFactory.php

```diff
@@ -14,16 +15,16 @@
  */
 final class NamespaceFactory
 {
-    private ClassNameMapper $classNameMapper;
+    private FinderInterface $finder;
 
-    public function __construct(private readonly CacheInterface $cache, ClassNameMapper|null $classNameMapper = null, private int|null $globTTL = 2)
+    public function __construct(private readonly CacheInterface $cache, FinderInterface|null $finder = null, private int|null $globTTL = 2)
     {
-        $this->classNameMapper = $classNameMapper ?? ClassNameMapper::createFromComposerFile(null, null, true);
+        $this->finder = $finder ?? new ComposerFinder();
     }
 
     /** @param string $namespace A PHP namespace */
-    public function createNamespace(string $namespace, bool $recursive = true): NS
+    public function createNamespace(string $namespace): NS
     {
-        return new NS($namespace, $this->cache, $this->classNameMapper, $this->globTTL, $recursive);
+        return new NS($namespace, $this->cache, clone $this->finder, $this->globTTL);
```

Comment:
I was cloning it here because of different namespaces called ->inNamespace on the same instance, I'm not sure how long running apps will be different but I think that's enough for NS class as $namespace is immutable after creation